### PR TITLE
Improve Test_Thread failure diagnostics

### PR DESCRIPTION
**Pull request by pshipton**
Related to #9915. The failure we're tracking re-occurred, but in a
different place. I've tried to over them all.

[ci skip]

Signed-off-by: Peter Shipton <Peter_Shipton@ca.ibm.com>

### CODE REVIEW

File: test/functional/Java8andUp/src/org/openj9/test/java/lang/Test_Thread.java [hunk 1]
```diff
@@ -163,6 +163,7 @@ public void test_Constructor2() {
 			ct = new Thread(new SimpleThread(10));
 			ct.start();
 		} catch (Exception e) {
+			e.printStackTrace();
 			AssertJUnit.assertTrue("Failed to create subthread", false);
```
Comment (keithc-ca) [inline test/functional/Java8andUp/src/org/openj9/test/java/lang/Test_Thread.java:167]: This sequence might better be handled by:
```
    Assert.fail("Failed to create subthread", e);
```
See [org.testng.Assert#fail](https://www.javadoc.io/static/org.testng/testng/6.8.1/org/testng/Assert.html#fail(java.lang.String,%20java.lang.Throwable))

Comment (pshipton) [reply test/functional/Java8andUp/src/org/openj9/test/java/lang/Test_Thread.java]: Updated

File: test/functional/Java8andUp/src/org/openj9/test/java/lang/Test_Thread.java [hunk 2]
```diff
@@ -787,6 +787,7 @@ public void test_join() {
 			// }
 			th.join();
 		} catch (InterruptedException e) {
+			AssertJUnit.fail("Interrupted", e);
 			result = false;
```
Comment (keithc-ca) [inline test/functional/Java8andUp/src/org/openj9/test/java/lang/Test_Thread.java:791]: The local variable `result` becomes useless with this change. Unless we expect to revert this later, the local variables should be removed (here and in `test_join2()` and `test_join3()`).

Comment (pshipton) [reply test/functional/Java8andUp/src/org/openj9/test/java/lang/Test_Thread.java]: Updated.

File: test/functional/Java8andUp/src/org/openj9/test/java/lang/Test_Thread.java [hunk 3]
```diff
@@ -787,10 +786,10 @@ public void test_join() {
 			// }
 			th.join();
 		} catch (InterruptedException e) {
-			result = false;
+			// killer.interrupt();
+			AssertJUnit.fail("Hung joining a non-started thread", e);
 		}
 		// killer.interrupt();
```
Comment (keithc-ca) [inline test/functional/Java8andUp/src/org/openj9/test/java/lang/Test_Thread.java:792]: What is the purpose of the comments referring to 'killer' (there is nothing available here by that name)?

Comment (pshipton) [reply test/functional/Java8andUp/src/org/openj9/test/java/lang/Test_Thread.java]: I couldn't say, just keeping it consistent with what was already there. 
I've removed the hopefully obsolete comments.

**Comment by keithc-ca** [comment]
Jenkins test sanity zlinux jdk11

File: test/functional/Java8andUp/src/org/openj9/test/java/lang/Test_Thread.java [hunk 4]
```diff
@@ -163,7 +163,7 @@ public void test_Constructor2() {
 			ct = new Thread(new SimpleThread(10));
 			ct.start();
 		} catch (Exception e) {
-			AssertJUnit.assertTrue("Failed to create subthread", false);
+			AssertJUnit.fail("Failed to create subthread", e);
```
Comment (keithc-ca) [inline test/functional/Java8andUp/src/org/openj9/test/java/lang/Test_Thread.java:166]: This needs to use `Assert.fail` like on line 276 below. This correction is required several other places below.

**Comment by pshipton** [comment]
Jenkins test sanity zlinux jdk11

File: test/functional/Java8andUp/src/org/openj9/test/java/lang/Test_Thread.java (hunk 4 above)
Comment (pshipton) [reply test/functional/Java8andUp/src/org/openj9/test/java/lang/Test_Thread.java]: Fixed, I hope.

**Pull request merged**